azure/services/roleassignments/roleassignments.go

```diff
@@ -18,132 +18,121 @@ package roleassignments
 
 import (
 	"context"
-	"fmt"
 
-	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/authorization/mgmt/authorization"
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-04-01/compute"
-	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/pkg/errors"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/scalesets"
 	"sigs.k8s.io/cluster-api-provider-azure/azure/services/virtualmachines"
+	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
-const azureBuiltInContributorID = "b24988ac-6180-42a0-ab88-20f7382dd24c"
+const serviceName = "roleassignments"
 
 // RoleAssignmentScope defines the scope interface for a role assignment service.
 type RoleAssignmentScope interface {
 	azure.ClusterDescriber
```

Comment:
is azure.ClusterDescriber still used anywhere? If not, let's remove it. If it is and only for one or two methods, can we list the specific methods needed in the interface instead of including `azure.ClusterDescriber` which is pretty broad?

Reply:
`ResourceGroup()` is what is used right now. 
I could have got rid of the `ClusterDescriber` in the `RoleAssignmentScope` interface but had to use due to the following reason:

- Reconcile RoleAssignment needs principal ID. To get the principal ID, an api call needs to be done which needs resource group. Now we can use `RoleAssignmentSpecs` to fetch the resource group but `RoleAssignmentSpecs` needs `PrincipalID`

Reply:
Can we just replace `azure.ClusterDescriber` by `ResourceGroup()` then? If we just need that one func might as well be explicit instead of including the entire `ClusterDescriber` interface which is pretty broad

Reply:
Done